src/app/invest/invest.component.ts

```diff
@@ -82,54 +51,117 @@ export class InvestComponent implements OnInit {
+
+        const projectInvestGap = expected - funded;
+        const userInvestGap = max - userInvested;
+
+        const userMinInvest = min > userInvested ? min : 1_00;
+        const userMaxInvest = Math.min(projectInvestGap, userInvestGap);
+
+        const maximumReached = max === userInvested;
+
+        return <InvestmentData>{
+            roi: project.roi,
+
+            projectExpected: expected,
+            projectMinPerTx: min,
+            projectMaxPerUser: max,
+            projectFunded: funded,
+            userBalance: userBalance,
+            userInvested: userInvested,
+
+            projectInvestGap: projectInvestGap,
+            userInvestGap: userInvestGap,
+
+            userMinInvest: userMinInvest,
+            userMaxInvest: userMaxInvest,
+
+            investDisabled: userMinInvest > userBalance || maximumReached,
+            notEnoughFunds: userMinInvest > userBalance,
+            maximumReached: maximumReached
+        };
+    }
+
+    private validAmount(control: AbstractControl): Observable<ValidationErrors | null> {
+        return combineLatest([this.investmentData$]).pipe(take(1),
+            map(([investmentData]) => {
+                    this.projectionSubject.next(null);
                     const amount = control.value;
-                    const amountInvested = this.investmentDetails.amountInvested;
 
-                    if (amount < this.minUserInvest) {
+                    if (amount < investmentData.userMinInvest) {
                         return {amountBelowMin: true};
-                    } else if (amount > this.maxUserInvest) {
+                    } else if (amount > investmentData.userMaxInvest) {
                         return {amountAboveMax: true};
-                    } else if (amount > project.expected_funding) {
-                        return {amountAboveExpFunding: true};
-                    } else if (amount > this.investmentDetails.walletBalance) {
+                    } else if (amount > investmentData.userBalance) {
                         return {amountAboveBalance: true};
                     } else {
-                        this.totalInvestment = amountInvested + amount;
-                        this.oneYearReturn = this.calculateReturn(this.totalInvestment, 1, project.roi.from, project.roi.to);
-                        this.fiveYearsReturn = this.calculateReturn(this.totalInvestment, 5, project.roi.from, project.roi.to);
-                        this.tenYearsReturn = this.calculateReturn(this.totalInvestment, 10, project.roi.from, project.roi.to);
+                        this.calculatePredictions(amount, investmentData);
                         return null;
                     }
                 }
             )
         );
     }
 
-    calculateReturn(totalInvestment: number, year: number, roiFrom: number, roiTo: number) {
-        return this.currencyPipe.transform((totalInvestment * (roiFrom * year)) / 100) + ' - '
```

Comment:
As Mislav told you, the predictions were 100 times greater than it should be. On that, you just divided the entire expression with 100, which gives the correct result, but semantically is **_wrong_** for anyone that comes after and see this.

When you see the entire expression divided by 100, you have no reason why it should be divided. It's better and much clearer to divide the variable which should be divided; in this case - `roiFrom`.

So instead of this:
`(totalInvestment * (roiFrom * year)) / 100`
It is healthier to write this:
`totalInvestment * roiFrom / 100 * year`
**And even more healthier** to write this:
`totalInvestment * (roiFrom / 100) * year`.

Use parentheses in expressions when:
1. you must because of the precedence of operators,
2. it is semantically much clearer to add them (in this case `(roiFrom / 100)` because `roiFrom` saved as a number, not a percent, and  ).

In your case (`(totalInvestment * (roiFrom * year)) / 100`), parentheses have no meaning. 
Be tidy when you write expressions to have a meaning for the reader, not only to do the job.